core/src/main/java/org/springframework/security/access/expression/AbstractSecurityExpressionHandler.java

```diff
@@ -102,22 +106,47 @@ protected abstract SecurityExpressionOperations createSecurityExpressionRoot(
 			Authentication authentication, T invocation);
 
 	protected RoleHierarchy getRoleHierarchy() {
+		if(! roleHierarchySet && context != null) {
```

Comment:
maybe extract this if sentence to a conditional method so that the code would be easier to read
e.g. private boolean roleHerarchyNotSetForValidContext(){
! roleHierarchySet && context != null
}